### PR TITLE
Add sync support

### PR DESCRIPTION
**Pull request by acbramley**
- Adds Sync CRD for co-ordinating a Backup and Restore process
- Sync waits for DeploymentConfig to be running before starting the restore.
- Restore operator already takes care of waiting for the Backup to complete

### CODE REVIEW

**Comment by pingers** [comment]
Is it common with Go projects to commit the vendor dir? Just curious.

**Comment by acbramley** [comment]
@pingers it is!

File: pkg/apis/extension/v1/sync_types_test.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	created := &Sync{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "foo",
+			Namespace: "default",
+		}}
+	g := gomega.NewGomegaWithT(t)
+
+	// Test Create
+	fetched := &Sync{}
+	g.Expect(c.Create(context.TODO(), created)).To(gomega.Succeed())
```
Comment (pingers) [inline pkg/apis/extension/v1/sync_types_test.go:42]: Where does "c" come from?

File: pkg/utils/k8s/sync/backup.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	extensionv1 "github.com/universityofadelaide/shepherd-operator/pkg/apis/extension/v1"
+)
+
+// Backup function to ensure the Object is in sync.
```
Comment (pingers) [inline pkg/utils/k8s/sync/backup.go:11]: Where are these used?

**Comment by pingers** [comment]
:shipit:

File: pkg/apis/extension/v1/sync_types_test.go (hunk 1 above)
Comment (acbramley) [reply pkg/apis/extension/v1/sync_types_test.go:42]: looks like it's a global in v1_suite_test.go - these tests are auto generated though

File: pkg/utils/k8s/sync/backup.go (hunk 2 above)
Comment (acbramley) [reply pkg/utils/k8s/sync/backup.go:11]: In the sync_controller when it's calling `CreateOrUpdate` on the backup

**Pull request merged**